docs/upgrading.md

```diff
@@ -0,0 +1,73 @@
+# Upgrading the Agent
```

Comment:
Please use sentence case "Upgrading the agent", unless you're referring to a specific agent such as the SignalFx Smart Agent or the SignalFx Smart Agent Receiver

Reply:
Based on what I've now read, rewrite this heading to be 

Upgrading the Splunk Distribution of OpenTelemetry Java